auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -275,6 +278,15 @@ private TypeSpec createTypeAdapter(ClassName className, TypeName autoValueClassN
       FieldSpec field = entry.getValue();
       names.add(prop.serializedName());
 
+      TypeVariableName usedDeclaredGenericTypeName = null;
+      if (genericTypeNames != null && prop.type instanceof TypeVariableName)
+        usedDeclaredGenericTypeName = (TypeVariableName) prop.type;
+
+      if (prop.type instanceof ParameterizedTypeName && genericTypeNames != null) {
+        usedDeclaredGenericTypeName = (TypeVariableName) ((ParameterizedTypeName) prop.type)
+            .typeArguments.get(0);
```

Comment:
yeah if you wouldn't mind. I don't think we want to only have partial support, and I think it should be doable to just match on all the type arguments